### ISSUE TITLE
Feature request: CModel change

### ISSUE THREAD

**Issue by bondjimbond** [mjordan/islandora_object_clone#6]
I haven't tried out this module, but it promises to be very valuable. It could also solve a problem that has not yet been addressed by core code: changing an object's content model.

Would it be possible to, as part of the cloning process, rewrite the RELS-EXT to assign a different content model? It might require some restrictions and checks to make it work, but it would be pretty nice to have.

**Comment by mjordan** [comment]
@bondjimbond yes, it would be, but as you say, unlike other relationships, safely redefining hasModel would require a bunch of preemptive checks. Could be done. Can we start listing the checks here in this issue and go from there? I'll start with:

* checking that the source object has all of the datastreams required by the target content model
* ensuring that the target collection allows the new content model
* allow a content model change for top-level objects only (book, newspaper issue) and not child pages in one go

**Comment by bondjimbond** [comment]
> checking that the source object has all of the datastreams required by the target content model

Not really required, but could the cloning process also generate any standard derivatives that aren't already present?

As to other checks...

- Checking validity of parent relationships? (You can't have a Large Image be isMemberOf a newspaper.) Alternatively - allow a select list of new parents?
- Checking the mimetype of the OBJ file (or PDF for some CModels) to ensure it's valid for the new CModel

**Comment by mjordan** [comment]
> could the cloning process also generate any standard derivatives that aren't already present?

That should happen automatically.

**Comment by mjordan** [comment]
> You can't have a Large Image be isMemberOf a newspaper.

I think this is just implied in Islandora, AFAIK there are no ways to enforce this.  Could be wrong about that.
